Add Ad Slot component

Jakeii opened the pull request:
Add an ad slot component that will tell the commercial bundle to place an ad here.

It uses a `sizeMapping` to decide what sizes to fill at what viewport size.
The breakpoints cascade from the smallest to the largest, AKA if only mobile and desktop are defined, the mobile size will be used for phablet and tablet.

njdehoog commented on src/lib/components/particles/ad-slot/ad-slot.stories.jsx:
```diff
@@ -0,0 +1,97 @@
+        desktop: desktopSizes ? desktopSizes.split('|').map((size) => size.split(',').map(Number)) : undefined,
+      }
+
+      console.log(sizeMapping)
+
+      const viewportWidth = window.innerWidth
+
+      console.log(viewportWidth)
```
Could we remove the two `console.log` statements here pls?

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+    content: 'Advertisement';
+  display: block;
+  position: relative;
+  font-family: GuardianTextSans, "Guardian Text Sans Web", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
```
We use the custom properties defined in font-families.css for this. In this case it would be `var(--text-sans)`.

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+  display: block;
+  position: relative;
+  font-family: GuardianTextSans, "Guardian Text Sans Web", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
+  font-size: 0.75rem;
```
rem sizes behave unexpectedly on the apps unfortunately, which is why we use px sizes

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+  position: relative;
+  font-family: GuardianTextSans, "Guardian Text Sans Web", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
+  font-size: 0.75rem;
+  line-height: 1.3;
```
Similarly, could you use `var(-sans-line-height)` here pls?

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+  --source-text-decoration-thickness: 2px;
+  height: 24px;
+  max-height: 24px;
+  background-color: #F6F6F6;
```
I wonder if this should support dark mode on the apps. In that case you could replace it with `var(--tertiary-bg-color)`

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+  max-height: 24px;
+  background-color: #F6F6F6;
+  padding: 0 8px;
+  border-top: 1px solid #DCDCDC;
```
And this with `var(--border-divider-color)`

njdehoog commented on src/lib/components/particles/ad-slot/stories.module.scss:
```diff
@@ -0,0 +1,22 @@
+  background-color: #F6F6F6;
+  padding: 0 8px;
+  border-top: 1px solid #DCDCDC;
+  color: #707070;
```
You get my point by now

Jakeii commented:
Thanks for the review! @njdehoog I've removed the logging and fixed the import.

THe styles are just for the storybook component, emulating the advert label which is added by the commercial bundle in production, so I think it's okay to have these hard coded values here?

njdehoog commented:
Ah yes, I didn't spot that. You're right
